src/docfx/lib/github/GitHubUserCache.cs

```diff
@@ -58,13 +58,22 @@ public async Task<(Error error, GitHubUser user)> GetByLogin(string login)
 
             var user = TryGetByLogin(login);
             if (user != null)
-                return user.IsValid() ? ((Error)null, user) : (Errors.GitHubUserNotFound(login), null);
+            {
+                user.Expiry = NextExpiry();
+                if (user.IsValid())
+                    return ((Error)null, user);
+                if (!user.IsPartial())
```

Comment:
>!user.IsPartial() [](start = 20, length = 17)

why return null if it's not partial?

Reply:
There is 3 status for user cache: valid user, non-existed user, and partial user. If it's not partial here, it must be a non-existed user, when should report a warning.

---
In reply to: [223551520](https://github.com/dotnet/docfx/pull/3467#discussion_r223551520) [](ancestors = 223551520)